lib/BuildSystem/BuildSystemFrontend.cpp

```diff
@@ -378,6 +409,11 @@ BuildSystemFrontendDelegate::createExecutionQueue() {
 
 void BuildSystemFrontendDelegate::cancel() {
   auto delegateImpl = static_cast<BuildSystemFrontendDelegateImpl*>(impl);
+  assert(delegateImpl->getStatus() == BuildSystemFrontendDelegateImpl::Status::Initialized);
```

Comment:
I think it is probably legal to call this with IntializationFailed, too, because client won't necessarily know at the time they call.